src/backend/dx11/src/lib.rs

```diff
@@ -3412,15 +3409,82 @@ impl Image {
     }
 }
 
-#[derive(Clone)]
 pub struct ImageView {
     subresource: UINT,
     format: format::Format,
-    rtv_handle: Option<ComPtr<d3d11::ID3D11RenderTargetView>>,
-    srv_handle: Option<ComPtr<d3d11::ID3D11ShaderResourceView>>,
-    dsv_handle: Option<ComPtr<d3d11::ID3D11DepthStencilView>>,
-    rodsv_handle: Option<ComPtr<d3d11::ID3D11DepthStencilView>>,
-    uav_handle: Option<ComPtr<d3d11::ID3D11UnorderedAccessView>>,
+    rtv_handle: Option<*mut d3d11::ID3D11RenderTargetView>,
+    srv_handle: Option<*mut d3d11::ID3D11ShaderResourceView>,
+    dsv_handle: Option<*mut d3d11::ID3D11DepthStencilView>,
+    rodsv_handle: Option<*mut d3d11::ID3D11DepthStencilView>,
+    uav_handle: Option<*mut d3d11::ID3D11UnorderedAccessView>,
+    borrowed: bool,
+}
+
+impl Clone for ImageView {
+    fn clone(&self) -> Self {
+        Self {
+            subresource: self.subresource,
+            format: self.format,
+            rtv_handle: self.rtv_handle.as_ref().map(|&p| {
```

Comment:
Why are we adding the references here? Or rather, the question is, why would we ever want to clone it while bumping the refcounts? Perhaps, there is only one clone implementation that needs to exist

Reply:
For a similar reason as why `Cow` turns owned when you clone it and `ComPtr` des; I want the default action to take references, so we don't end up with dangling pointers if at all possible. Additionally, if we do own it, we need to clone to get another owned `ImageView`.

That is a long-winded way of saying: Clone not taking references would be surprising, and I don't want to be surprising with the potential for dangling references.

Reply:
These are raw pointers (unlike Cow/ComPtr). When you Clone or Copy a raw pointer, you don't get surprised. You can implement this as just a `copy_raw()` method instead of `impl Clone` if that makes it cleaner.
The main question still stands - why would we need both borrowing copy and owning copy?

Reply:
Let me see what happens if I make clone only borrowing and if nothing explodes that'll be fine with me.